integration-tests/bin/load-integration-data.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+## Use the Environment var or the default
+if [[ -z "${ELASTIC_SEARCH_HOST}" ]]; then
+    ES_HOST="http://localhost:9200"
+else
+    ES_HOST=${ELASTIC_SEARCH_HOST}
+fi
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+## Wait until docker is up.
+echo "Waiting for ES Service at ${ES_HOST} to Start"
+until $(curl --output /dev/null --silent --head --fail "${ES_HOST}"); do
+    printf '.'
+    sleep 1
+done
+echo "ES Service is up"
+
+# First wait for ES to start...
+response=$(curl --write-out %{http_code} --silent --output /dev/null "${ES_HOST}")
+
+until [ "$response" = "200" ]; do
+    response=$(curl --write-out %{http_code} --silent --output /dev/null "${ES_HOST}")
+    >&2 echo "Elastic Search is unavailable - sleeping"
+    sleep 1
+done
+
+# next wait for ES status to turn to Green
+health_check="curl -fsSL ${ES_HOST}/_cat/health?h=status"
+health=$(eval $health_check)
+echo "Waiting for ES status to be ready"
+until [[ "$health" = 'green' ]]; do
+    >&2 echo "Elastic Search is not ready - sleeping"
+    sleep 10
+    health=$(eval $health_check)
+done
+echo "ES status is $health"
+
+echo "Loading the index mappings"
+pushd $DIR/../data/source >/dev/null
+HDR="Content-Type: application/json"
+response=$(curl -s -H "$HDR" -XPUT ${ES_HOST}/cgov -d @cgov-mapping.json)
+if [[ ! $response =~ acknowledged[^a-zA-Z]+true ]]; then
+    echo Failure loading search mappings
+    echo $response
+fi
+response=$(curl -s -H "$HDR" -XPUT ${ES_HOST}/autosg -d @autosg-mapping.json)
+if [[ ! $response =~ acknowledged[^a-zA-Z]+true ]]; then
+    echo Failure loading autosuggest mappings
+    echo $response
+fi
+
+echo "Loading the records"
+response=$(curl -s -H "$HDR" -XPOST ${ES_HOST}/_bulk --data-binary @cgov-data-1.jsonl)
+if [[ $response =~ errors[^a-zA-Z]+true ]]; then
+    echo Failure loading search records
+    echo $response
+fi
+response=$(curl -s -H "$HDR" -XPOST ${ES_HOST}/_bulk --data-binary @cgov-data-2.jsonl)
+if [[ $response =~ errors[^a-zA-Z]+true ]]; then
+    echo Failure loading search records
+    echo $response
+fi
+response=$(curl -s -H "$HDR" -XPOST ${ES_HOST}/_bulk --data-binary @cgov-data-3.jsonl)
+if [[ $response =~ errors[^a-zA-Z]+true ]]; then
+    echo Failure loading search records
+    echo $response
+fi
```

Comment:
Is it possible to put these into a single file without violating GitHub's maximum file size restriction?  They were split up on the ticket because the size limit on tickets is smaller than the repository's code fils.

Reply:
As I understand it, the limit on GitHub is 100MB, and the total for these three is well over that. I considered getting around that by compressing them, but that would have assumed the availability of the decompression tool on the host machine (remember, we're not running the tests in a container). But that all became moot when I discovered that ElasticSearch can't handle that much data in a bulk load. That's what I was trying to tell you on Monday.

![image](https://user-images.githubusercontent.com/1634628/123340081-a5f30d00-d519-11eb-94e9-5f711a4c8382.png)

Reply:
Ah, I hadn't caught that you'd discovered the bulk load wouldn't handle it.  [The list of installed software](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-software) says there are "unzip" tools available, but it appears to be different tools for each OS.

Let's leave the multiple files then, but  please add a comment to the script explaining that combining the files would cause bulk load to fail.

Reply:
Comment added.

> I hadn't caught that you'd discovered the bulk load wouldn't handle it.

That's why I wrote "... the fact that if you do a _bulk load with the entire set ES just does nothing silently failing" instead of "... the possibility that ...." 😉